crypto/src/varint.rs

```diff
@@ -72,30 +95,35 @@ macro_rules! from_uint(
 from_uint!(u8, u16, u32, u64, u128);
 
 impl VarUint {
-    /// Construct [`VarUint`].
-    pub fn new(bytes: impl AsRef<[u8]>) -> Result<Self> {
+    /// Construct `VarUint`.
+    pub fn new(bytes: impl AsRef<[u8]>) -> Result<Self, ConvertError> {
         let idx = bytes
             .as_ref()
             .iter()
             .enumerate()
             .find(|&(_, &byte)| (byte & 0b1000_0000) == 0)
-            .ok_or_else(|| eyre!("Last byte should be less than 128"))?
+            .ok_or_else(|| ConvertError::new("Last byte should be less than 128"))?
             .0;
         let (payload, empty) = bytes.as_ref().split_at(idx + 1);
         let payload = payload.to_vec();
 
         if empty.is_empty() {
-            Ok(Self { payload })
-        } else {
-            Err(eyre!("Last byte shouldn't be followed by anything"))
+            return Ok(Self { payload });
         }
+
+        Err(ConvertError::new(
+            "Last byte shouldn't be followed by anything",
```

Comment:
Could also print what follows the last byte, but I don't insist.

Reply:
that's a good point, though I'd have to move from `&'static str` to `String`. Will add this